src/ReasonReactErrorBoundary.re

```diff
@@ -11,32 +11,60 @@ type params('error) = {
   info,
 };
 
-[%%mel.raw
-  {|
-  var React = require("react");
-  var ErrorBoundary = (function (Component) {
-    function ErrorBoundary(props) {
-      Component.call(this);
-      this.state = {error: undefined};
-    };
-    ErrorBoundary.prototype = Object.create(Component.prototype);
-    ErrorBoundary.prototype.componentDidCatch = function(error, info) {
-      this.setState({error: {error: error, info: info}});
-    };
-    ErrorBoundary.prototype.render = function() {
-      return this.state.error != undefined
-        ? this.props.fallback(this.state.error)
-        : this.props.children
-    };
-    return ErrorBoundary;
-  })(React.Component);
-|}
-];
+[@mel.scope "Object"] external objCreate: 'a => Js.t({..}) = "create";
+
+[@ocaml.warning "-69"]
+type state('error) = {error: params('error)};
+type reactComponentClass;
+
+[@mel.module "react"] external component: reactComponentClass = "Component";
+[@mel.send] external componentCall: (reactComponentClass, _) => unit = "call";
+
+type componentPrototype;
+[@mel.get]
+external componentPrototype: reactComponentClass => componentPrototype =
+  "prototype";
+
+[@ocaml.warning "-21"]
```

Comment:
I trend to add the why when I disable some warnings, since 21/69 isn't descriptive enough

Reply:
good idea! I realized we don't actually need these at all. check a8d20ef2c6acd2334a5597b705402bef6c14eb9f